src/io/leopard/toLeopard.ts

```diff
@@ -69,6 +69,215 @@ const JS_RESERVED_WORDS = [
   "with"
 ];
 
+/**
+ * Global identifiers which Leopard sprites (subclasses of `Sprite`) must not
+ * be named as, since that would overwrite or be a reference binding error.
+ *
+ * Only capitalized identifiers need to be listed here: generated sprite names
+ * will never conflict with identifiers whose first letter is lowercase.
+ * (This is also why JS reserved words aren't listed here - they're all
+ * lowercase, so don't conflict with generated sprite names.)
+ *
+ * However, it *must* include even (capitalized) identifiers which *aren't*
+ * provided by Leopard, if any part of generated Leopard code could directly
+ * refer to those identifiers (expecting some browser-provided value, rather
+ * than a generated `Sprite` subclass!).
+ */
+const LEOPARD_RESERVED_SPRITE_NAMES = [
+  // Browser-provided identifiers
+  "Date",
+  "Math",
```

Comment:
**Question:** Do we want to blacklist browser-provided identifiers like `Array`, `Audio`, `Error`, `File`, `Map`, `Set`, `Temporal`, etc., which are not used in generated Leopard code but could cause conflicts or confusion when the user starts adding their own hand-written JS code?

There are a lot of them. And keeping up-to-date might be challenging.

Reply:
Thanks, this is a good point! We've followed it up in detail here: #138